docs/features/software-catalog/descriptor-format.md

```diff
@@ -527,6 +529,41 @@ The current set of well-known and common values for this field is:
   [Protocol Buffers](https://developers.google.com/protocol-buffers) to use with
   [gRPC](https://grpc.io/).
 
+### `spec.lifecycle` [required]
```

Comment:
Wondering if it might make more sense to have spec.lifecycle optional and any logic would fallback the component's `lifecycle` that implements the API.